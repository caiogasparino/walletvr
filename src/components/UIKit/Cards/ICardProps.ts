export interface ICardProps {
  cardName?: string;
  name?: string;
  cardNumber?: string;
  cardDate?: string;
  cardType: string;
  opacity?: number;
  testId?: string;
}
