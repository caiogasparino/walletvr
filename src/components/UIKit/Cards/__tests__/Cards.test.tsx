import React from 'react';
import {render} from '@testing-library/react-native';
import {Card} from '@UIKit';
import {ICardProps} from '../ICardProps';

const props: ICardProps = {
  cardName: 'Visa',
  name: 'John Doe',
  cardNumber: '1234 5678 9012 3456',
  cardDate: '10/20',
  cardType: 'Black Card',
  opacity: 1,
};

test('renders Card component correctly', () => {
  const {toJSON} = render(<Card {...props} />);

  expect(toJSON()).toMatchSnapshot();
});
