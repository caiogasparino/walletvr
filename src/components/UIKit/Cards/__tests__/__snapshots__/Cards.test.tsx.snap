// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card component correctly 1`] = `
<View
  cardType="Black Card"
  opacity={1}
  style={
    {
      "backgroundColor": "#111111",
      "borderBottomLeftRadius": 16,
      "borderBottomRightRadius": 16,
      "borderTopLeftRadius": 16,
      "borderTopRightRadius": 16,
      "gap": 10,
      "height": 180,
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 16,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "shadowColor": "rgba(0, 0, 0, 0.25)",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 12,
      "width": 300,
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "display": "flex",
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
          "display": "flex",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "PTSansCaption-Regular",
            "fontSize": 18,
            "fontWeight": undefined,
            "lineHeight": 20,
          }
        }
      >
        Visa
      </Text>
    </View>
    <View
      paddingBottom={10}
      style={
        {
          "borderBottomColor": "transparent",
          "borderBottomWidth": 0,
          "borderLeftColor": "transparent",
          "borderRightColor": "transparent",
          "borderTopColor": "transparent",
          "paddingBottom": 10,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "flex-start",
          "display": "flex",
          "gap": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "PTSansCaption-Regular",
            "fontSize": 16,
            "fontWeight": undefined,
            "lineHeight": 18,
          }
        }
      >
        John Doe
      </Text>
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "PTSansCaption-Regular",
            "fontSize": 14,
            "fontWeight": undefined,
            "lineHeight": 16,
          }
        }
      >
        •••• •••• •••• 3456
      </Text>
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "PTSansCaption-Regular",
            "fontSize": 14,
            "fontWeight": undefined,
            "lineHeight": 16,
          }
        }
      >
        10/20
      </Text>
    </View>
  </View>
</View>
`;
